### DISCUSSION TITLE
PostgreSQL: Error: Table name xxxx specified more than once

### DISCUSSION THREAD

**Discussion by kpturner** [Budibase/budibase#10215] [Help] [answered]
**Checklist**
- [x] I have searched budibase discussions and github issues to check if my issue already exists

**Hosting**
- Cloud
  - Tenant ID: monito

**Describe the bug**
I have two postgreSQL tables:

**country**
id_country (autoincrement PK)
id_currency (foreign key to currency table)
label_country (selected as display field for table)

**currency**
id_currency (autoincrement PK)
label_currency (selected as display field for table)

If I create two relationships between the two fiels everything stops working.

**To Reproduce**
I created a relationship between the two:
Type: One to many
From table "currency"
Primary key (currency): id_currency
To table "country"
Foreign key (country): id_currency
From table column name: currency
To table column name: country

At this point everything works as expected. I can edit a country and select the currency from a list of currencies.

Now additionally I have a many to many relationship between the two tables.  This is because a country can have additional currencies as well as its default.  So the **id_currency** on **country** is the default currency, but there may be additional to define.  For this there is another table:

**currency_country**
id_country (FK to country)
id_currency (FK to currency)

I defined a relationship to represent this:
Type: Many to many
From table: currency
To table: country
Through: currency_country
Foreign key (currency): id_currency
Foreign key (country): id_country
From table column name: currencies
To table column name: countries

Now when I try to display the data in the country file I get an error:  
_error: table name "currency" specified more than once_
 
When I display the data in the currency file I get:
_error: table name "country" specified more than once_

If I delete either of the relationships, the remaining one works as expected. 

So is it not possible to do what I want, or have I misconfigured it?

**Expected behavior**
This should be possible but it is not obvious how to do it without getting errors. 

**Desktop (please complete the following information):**
 - OS: iOS
 - Browser: Chrome
 - Version: 111

**Comment by ConorWebb96** [comment, marked as answer]
Hey @kpturner I've gone ahead and created this as an issue for you. It can be found here: https://github.com/Budibase/budibase/issues/10218 

I will close this discussion in the meantime, thank you for reporting this.

**Comment by kpturner** [comment]
Thankyou

 

From: Conor Webb ***@***.***>
Reply to: Budibase/budibase ***@***.***>
Date: Wednesday, 5 April 2023 at 13:24
To: Budibase/budibase ***@***.***>
Cc: Kevin Turner ***@***.***>, Mention ***@***.***>
Subject: Re: [Budibase/budibase] PostgreSQL: Error: Table name xxxx specified more than once (Discussion #10215)

 

Hey @kpturner I've gone ahead and created this as an issue for you. It can be found here: #10218

I will close this discussion in the meantime, thank you for reporting this.

**Comment by shogunpurple** [comment]
hi @kpturner - if possible, could you please share your postgres schema or a dump so that we can debug internally? This would help us out a lot with regards to the postgres issues you have been posting. If it's private, you can send the schema to `support@budibase.com` directly. Thank you!

**Reply by kpturner** [reply]
OK I will email you a SQL export
